src/common/POLFields/FieldsLocation/FieldsLocation.js

```diff
@@ -42,14 +50,16 @@ const FieldsLocation = ({ locations, disabled, isDisabledToChangePaymentInfo, wi
       renderField={(field) => (
         <Row>
           <Col xs={6}>
-            <FieldSelect
-              dataOptions={locations}
-              fullWidth
-              label={<FormattedMessage id="ui-orders.location.nameCode" />}
+            <FieldLocation
+              isDisabled={disabled}
+              isReduxField
+              labelId="ui-orders.location.nameCode"
+              locationsForDict={locations}
               name={`${field}.locationId`}
+              onChange={({ id }) => dispatch(change(`${field}.locationId`, id))}
```

Comment:
IMHO, it makes not too much sense to pass several props for only one line of using them. You can define your handler at level where you have them and pass only that callback.
second point - we don't use inline callbacks in functional components, if it's not clear why - please ping me, I'll describe